### PR TITLE
when onbuild is blocked, tolerate onbuild images but don't docker build them

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
@mfojtik ptal

File: pkg/build/strategies/layered/layered.go [hunk 1]
```diff
@@ -126,6 +127,10 @@ func (builder *Layered) SourceTar(config *api.Config) (io.ReadCloser, error) {
 
 //Build handles the `docker build` equivalent execution, returning the success/failure details
 func (builder *Layered) Build(config *api.Config) (*api.Result, error) {
+	if config.HasOnBuild && config.BlockOnBuild {
+		return nil, fmt.Errorf("builder image uses ONBUILD instructions but ONBUILD is blocked.")
```
Comment (mfojtik) [inline pkg/build/strategies/layered/layered.go]: "is blocked" sounds pretty hard to me :-) I would say "is not allowed" ?

File: pkg/api/types.go [hunk 2]
```diff
@@ -199,6 +199,9 @@ type Config struct {
 	// the container for images that don't have a tar binary available, if the
 	// image contains ONBUILD commands that would be executed.
 	BlockOnBuild bool
+
+	// HasOnBuild will be set to true if the builder image contains ONBUILD instructions
+	HasOnBuild bool
```
Comment (mfojtik) [inline pkg/api/types.go]: I think in the future we should stop bloating the Config and refactor the code ;-) But I'm OK wit this for now.

**Comment by mfojtik** [comment]
@bparees one rephrasing suggestion, otherwise LGTM

File: pkg/build/strategies/layered/layered.go (hunk 1 above)
Comment (bparees) [inline pkg/build/strategies/layered/layered.go]: updated.

File: pkg/api/types.go (hunk 2 above)
Comment (bparees) [inline pkg/api/types.go]: yeah, i didn't feel great about it, but it's sort of part of the config :)

**Comment by bparees** [comment]
[merge]

**Pull request merged**